ChaosMod/Components/EffectDispatcher.cpp

```diff
@@ -566,6 +569,16 @@ void EffectDispatcher::ClearMostRecentEffect()
 	}
 }
 
+std::vector<RegisteredEffect*> EffectDispatcher::GetRecentEffects(int distance)
+{
+	std::vector<RegisteredEffect*> temp;
+	for (int i = m_rgDispatchedEffectsLog.size() - 1; i --> distance;)
```

Comment:
Please just do `i > distance; i--` instead, it's going to be more clearer to readers.

Also make sure to check if you haven't reached the end of the list (i < 0)